### PR TITLE
docs: Add OpenDAL

### PR DESCRIPTION
**Pull request by Xuanwo**
Add OpenDAL into the list.

### CODE REVIEW

**Comment by palfrey** [comment]
This is failing due to an issue with accepted dash patterns. https://github.com/rust-unofficial/awesome-rust/pull/1755 solves this, but you'll need to merge main.

Also the image should have alt text, see the markdown error.

File: README.md [hunk 1]
```diff
@@ -1381,12 +1381,22 @@ See also [About Rust’s Machine Learning Community](https://medium.com/@autumn_
 
 [[filesystem](https://crates.io/keywords/filesystem)]
 * Operations
+<<<<<<< patch-1
```
Comment (palfrey) [inline README.md:1384]: This is a bad merge and needs fixing

Comment (Xuanwo) [reply README.md:1384]: Oh, sorry, my bad.

Comment (Xuanwo) [reply README.md]: Should be fixed, please review again.

**Pull request merged**